### PR TITLE
Convert zwave-node-config to lit/ts

### PR DESCRIPTION
**Pull request by cgarwood**
This was a lot more complex than the other config panel elements that I've converted to Lit/Typescript. Originally tried to do a mostly straight conversion and maintain the original logic, but ended up rewriting more than I anticipated originally, but I think it's a little cleaner now than it was originally.

Includes translations for everything except the actual configuration parameter titles/values and help text, as they come from OpenZWave .

### CODE REVIEW

File: src/panels/config/zwave/zwave-node-config.ts [hunk 1]
```diff
@@ -0,0 +1,390 @@
...
+  @property() private _configItem?: ZWaveConfigItem;
+  @property() private _wakeupInput: number = -1;
+  @property() private _selectedConfigParameter: number = -1;
+  @property() private _selectedConfigValue: number | string = -1;
+
+  protected render(): TemplateResult | void {
+    return html`
+      <div class="content">
+        <ha-card
+          header=${this.hass!.localize(
```
Comment (balloob) [inline src/panels/config/zwave/zwave-node-config.ts]:
```suggestion
          .header=${this.hass!.localize(
```

File: src/panels/config/zwave/zwave-node-config.ts [hunk 2]
```diff
@@ -0,0 +1,390 @@
...
+            "ui.panel.config.zwave.node_config.header"
+          )}
+        >
+          ${this.nodes[this.selectedNode].attributes.hasOwnProperty(
```
Comment (balloob) [inline src/panels/config/zwave/zwave-node-config.ts]: `"wake_up_interval" in this.nodes[this.selectedNode].attributes`

File: src/panels/config/zwave/zwave-node-config.ts [hunk 3]
```diff
@@ -0,0 +1,390 @@
...
+            "wake_up_interval"
+          )
+            ? html`
+                <div class="card-actions">
+                  <paper-input
+                    float-label="${this.hass!.localize(
```
Comment (balloob) [inline src/panels/config/zwave/zwave-node-config.ts]:
```suggestion
                    .floatLabel="${this.hass!.localize(
```

File: src/panels/config/zwave/zwave-node-config.ts [hunk 4]
```diff
@@ -0,0 +1,390 @@
...
+                      "ui.panel.config.zwave.common.wakeup_interval"
+                    )}"
+                    type="number"
+                    .value=${this._wakeupInput !== -1
+                      ? this._wakeupInput
+                      : this.hass!.localize(
+                          "ui.panel.config.zwave.common.unknown"
+                        )}
+                    @value-changed=${this._onWakeupIntervalChanged}
+                    placeholder=${this.nodes[this.selectedNode].attributes
```
Comment (balloob) [inline src/panels/config/zwave/zwave-node-config.ts]: Any value that's dynamic and is set on a web component, set the property.

```suggestion
                    .placeholder=${this.nodes[this.selectedNode].attributes
```

File: src/panels/config/zwave/zwave-node-config.ts [hunk 5]
```diff
@@ -0,0 +1,390 @@
...
+                      .wake_up_interval
+                      ? this.nodes[this.selectedNode].attributes
+                          .wake_up_interval
+                      : this.hass!.localize(
+                          "ui.panel.config.zwave.common.unknown"
+                        )}
+                  >
+                    <div suffix="">
```
Comment (balloob) [inline src/panels/config/zwave/zwave-node-config.ts]: Leftover from conversion

```suggestion
                    <div suffix>
```

File: src/panels/config/zwave/zwave-node-config.ts [hunk 6]
```diff
@@ -0,0 +1,390 @@
...
+                    ${this.hass!.localize(
+                      "ui.panel.config.zwave.node_config.set_wakeup"
+                    )}
+                  </ha-call-service-button>
+                </div>
+              `
+            : ""}
+          <div class="device-picker">
+            <paper-dropdown-menu
+              label=${this.hass!.localize(
```
Comment (balloob) [inline src/panels/config/zwave/zwave-node-config.ts]:
```suggestion
              .label=${this.hass!.localize(
```

File: src/panels/config/zwave/zwave-node-config.ts [hunk 7]
```diff
@@ -0,0 +1,390 @@
...
+                      ${state.key}: ${state.value.label}
+                    </paper-item>
+                  `
+                )}
+              </paper-listbox>
+            </paper-dropdown-menu>
+          </div>
+          ${this._configItem
+            ? html`
+                ${["List"].includes(this._configItem.value.type)
```
Comment (balloob) [inline src/panels/config/zwave/zwave-node-config.ts]:
```suggestion
                ${this._configItem.value.type === "List"
```

File: src/panels/config/zwave/zwave-node-config.ts [hunk 8]
```diff
@@ -0,0 +1,390 @@
...
+                  ? html`
+                      <div class="device-picker">
+                        <paper-dropdown-menu
+                          label=${this.hass!.localize(
```
Comment (balloob) [inline src/panels/config/zwave/zwave-node-config.ts]:
```suggestion
                          .label=${this.hass!.localize(
```

File: src/panels/config/zwave/zwave-node-config.ts [hunk 9]
```diff
@@ -0,0 +1,390 @@
...
+                            "ui.panel.config.zwave.node_config.config_value"
+                          )}
+                          dynamic-align
+                          class="flex"
+                          placeholder=${this._configItem.value.data}
```
Comment (balloob) [inline src/panels/config/zwave/zwave-node-config.ts]:
```suggestion
                          .placeholder=${this._configItem.value.data}
```

File: src/panels/config/zwave/zwave-node-config.ts [hunk 10]
```diff
@@ -0,0 +1,390 @@
...
+                          </paper-listbox>
+                        </paper-dropdown-menu>
+                      </div>
+                    `
+                  : ""}
+                ${["Byte", "Short", "Int"].includes(this._configItem.value.type)
+                  ? html`
+                      <div class="card-actions">
+                        <paper-input
+                          label=${this._configItem.value.data_items}
```
Comment (iantrich) [inline src/panels/config/zwave/zwave-node-config.ts]:
```suggestion
                          .label=${this._configItem.value.data_items}
```

File: src/panels/config/zwave/zwave-node-config.ts [hunk 11]
```diff
@@ -0,0 +1,390 @@
...
+                          type="number"
+                          value=${this._configItem.value.data}
```
Comment (iantrich) [inline src/panels/config/zwave/zwave-node-config.ts]:
```suggestion
                          .value=${this._configItem.value.data}
```

File: src/panels/config/zwave/zwave-node-config.ts [hunk 12]
```diff
@@ -0,0 +1,390 @@
...
+                          max=${this._configItem.value.max}
```
Comment (iantrich) [inline src/panels/config/zwave/zwave-node-config.ts]: etc for all dynamic here

File: src/panels/config/zwave/zwave-node-config.ts [hunk 13]
```diff
@@ -0,0 +1,390 @@
...
+                        >
+                        </paper-input>
+                      </div>
+                    `
+                  : ""}
+                ${["Bool", "Button"].includes(this._configItem.value.type)
+                  ? html`
+                      <div class="device-picker">
+                        <paper-dropdown-menu
+                          label=${this.hass!.localize(
```
Comment (iantrich) [inline src/panels/config/zwave/zwave-node-config.ts]:
```suggestion
                          .label=${this.hass!.localize(
```

File: src/panels/config/zwave/zwave-node-config.ts [hunk 14]
```diff
@@ -0,0 +1,390 @@
...
+          -webkit-flex: 1;
+          flex: 1;
+          -webkit-flex-basis: 0.000000001px;
+          flex-basis: 0.000000001px;
+        }
+      `,
+    ];
+  }
+
+  protected firstUpdated(changedProps): void {
```
Comment (iantrich) [inline src/panels/config/zwave/zwave-node-config.ts]:
```suggestion
  protected firstUpdated(changedProps: PropertyValues): void {
```

File: src/panels/config/zwave/zwave-node-config.ts [hunk 15]
```diff
@@ -0,0 +1,390 @@
...
+  }
+
+  protected updated(changedProps: PropertyValues): void {
+    super.updated(changedProps);
+    if (changedProps.has("selectedNode")) {
+      this._nodesChanged();
+    }
+  }
+
+  private serviceCalled(ev) {
```
Comment (iantrich) [inline src/panels/config/zwave/zwave-node-config.ts]:
```suggestion
  private serviceCalled(ev): void {
```

File: src/panels/config/zwave/zwave-node-config.ts [hunk 16]
```diff
@@ -0,0 +1,390 @@
...
+    if (ev.detail.success) {
+      setTimeout(() => {
+        this._refreshConfig(this.selectedNode);
+      }, 5000);
+    }
+  }
+
+  private _nodesChanged() {
```
Comment (iantrich) [inline src/panels/config/zwave/zwave-node-config.ts]:
```suggestion
  private _nodesChanged(): void {
```

File: src/panels/config/zwave/zwave-node-config.ts [hunk 17]
```diff
@@ -0,0 +1,388 @@
...
+                      <div class="device-picker">
+                        <paper-dropdown-menu
+                          .label=${this.hass!.localize(
+                            "ui.panel.config.zwave.node_config.config_value"
+                          )}
+                          class="flex"
+                          dynamic-align
+                          .placeholder=${this._configItem.value.data}
+                        >
+                          <paper-listbox
```
Comment (balloob) [inline src/panels/config/zwave/zwave-node-config.ts]: Why not use a checkbox?

Comment (cgarwood) [reply src/panels/config/zwave/zwave-node-config.ts]: It was a dropdown originally so I just left it like that.

**Pull request merged**